### PR TITLE
Display upcoming COVID status changes on the results page

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/7sslimxM
Supersedes: #5728

# What's changed?
Displays a warning on the Check Travel During Coronavirus results page if the COVID status of a country is changing.

The format of the Coronavirus travel data YAML file has also changed to support programmatically updating the COVID status of a world location.

# Why?
The World Location API won't contain COVID status information in time for the MVP smartanswer to be released, so we need to be able to get up to date status information from the YAML file. Updating the format of the YAML in this ways allows status changes to update automatically, and avoid the need for late night deployments.

# Expected changes

[Rendered version](https://smart-answer-automate-c-yfzsvn.herokuapp.com/check-travel-during-coronavirus/results?any_other_countries_1=yes&any_other_countries_2=yes&any_other_countries_3=no&going_to_countries_within_10_days=yes&transit_countries%5B%5D=none&travelling_with_children%5B%5D=none&vaccination_status=vaccinated&which_1_country=italy&which_2_country=south-africa&which_country=afghanistan)

|Before|After|
|:------|:----|
|Screenshot 2022-01-20 at 17 09 09|Screenshot 2022-01-21 at 17 02 21|

## Mobile view
Screenshot 2022-01-21 at 17 45 12

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/models/world_location.rb [hunk 1]
```diff
@@ -67,14 +67,50 @@ def self.travel_rules
...
+      start_date = Time.zone.parse(status["covid_status_applies_at"])
+      start_date.past?
+    end
+
+    return if current_statuses.blank?
+
+    current_statuses.max_by { |status| status["covid_status_applies_at"] }
+  end
+
+  def next_covid_status_data
```
Comment (Rosa-Fox) [inline app/models/world_location.rb:94]: These all look good. I'm wondering if we should move the initialiser up to above the instance methods to be more conventional?

File: test/flows/check_travel_during_coronavirus_flow_test.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require "test_helper"
 require "support/flow_test_helper"
 
-class CheckTravelDuringCoronavirusTest < ActiveSupport::TestCase
+class CheckTravelDuringCoronavirusFlowTest < ActiveSupport::TestCase
```
Comment (Rosa-Fox) [inline test/flows/check_travel_during_coronavirus_flow_test.rb:4]: 👍

File: test/unit/world_location_test.rb [hunk 3]
```diff
@@ -275,25 +275,66 @@ class WorldLocationTest < ActiveSupport::TestCase
...
+      should "find the next covid status for a location" do
+        assert_equal "not_red", @location.next_covid_status
+      end
+
+      should "find the next covid status applies at date for a location" do
+        assert_equal Time.zone.parse(@future_status_date), @location.next_covid_status_applies_at
+      end
+    end
+
+    context "covid statuses do not exist" do
```
Comment (Rosa-Fox) [inline test/unit/world_location_test.rb:321]: 👍

**Review by Rosa-Fox — APPROVED** [review]
Looks good. I've tested manually. I had one comment but its not blocking.

File: app/models/world_location.rb (hunk 1 above)
Comment (leenagupte) [reply app/models/world_location.rb:102]: Oops! Will move.

**Comment by leenagupte** [comment]
@Rosa-Fox Thanks for your review. I've moved the initialize method to the top of the instance methods where it should be.

**Pull request merged**